docs/tinymist/frontend/neovim.typ

```diff
@@ -22,121 +26,81 @@ See #link("https://github.com/Myriad-Dreamin/tinymist#features")[Tinymist Featur
-
-Then, check and restart Neovim.
-
-=== Setup for #link("https://github.com/AstroNvim")[AstroVim]
-
-AstroNvim now uses tinymist by default. Please check the #link("https://github.com/AstroNvim/astrocommunity/tree/main/lua/astrocommunity/pack/typst")[setup script].
-
-=== Setup for #link("https://github.com/neoclide/coc.nvim")[coc.nvim]
-
-You can edit the `coc-settings.json` by executing `:CocConfig`:
-
-```json
-{
-  "languageserver": {
-    "tinymist": {
-      "command": "tinymist",
-      "filetypes": ["typst"],
-      "settings": { ... }
+- Or manually:
+
+ #include "common-finding-executable.typ"
+
+== Configuration
+- With `lspconfig` 
+    ```lua
+    require("lspconfig")["tinymist"].setup {
+        settings = {
+            tinymist = {
+                settings = {
+                    formatterMode = "typstyle",
+                    exportPdf = "onType",
+                    semanticTokens = "disable"
+                    -- ...
+                },
+            },
+        },
     }
-  }
-}
-```
-
-== Tips
-<tips>
-=== Working with Multiple-File Projects
-<working-with-multiple-file-projects>
-The solution is a bit internal, which should get further improvement, but you can pin a main file by command.
-
-```lua
--- pin the main file
-vim.lsp.buf.execute_command({ command = 'tinymist.pinMain', arguments = { vim.api.nvim_buf_get_name(0) } })
--- unpin the main file
-vim.lsp.buf.execute_command({ command = 'tinymist.pinMain', arguments = { nil } })
-```
+    ```
+
+- Or with `Coc.nvim`
+
+    ```json
+    {
+      "languageserver": {
+        "tinymist": {
+          "command": "tinymist",
+          "filetypes": ["typst"],
+          "settings": { ... }
+        }
+      }
+    }
+    ```
+- Or finally with the builtin lsp protocol
+    
+    ```lua
+    vim.lsp.config['tinymist'] = {
+        cmd = {'tinymist'}, 
+        filetypes = {'typst'}
+        settings = {
+            -- ...
+        }
+    }
+    ```
+    For a full list of availible settings see #link("https://github.com/Myriad-Dreamin/tinymist/tree/main/editors/neovim/Configuration.md")[Tinymist Server Configuration]. 
 
-There is also a plan to support multiple-file projects with a workspace configuration, but I don’t know whether it is Neovim’s way, so it needs further discussion.
+== Formatting
+    Install either `typststyle` or `typstfmt` either via source or via `Mason`. Then specify the formatter in your config
 
-#note-box[
-  `tinymist.pinMain` is a stateful command, and tinymist doesn't remember it between sessions (closing and opening the editor).
-]
+    ```lua 
+    formatterMode = "typstyle"
+    ```
+== Preview
+    There is work in progress to integrate #link("https://github.com/chomosuke/typst-preview.nvim")[typst-preview.nvim] directly into `tinymist`. In the meantime you can find can find installation and configuration instructions on that repo. This should be a shortterm solution. 
```

Comment:
I believe we have introduced an announcement here that we will provide some plugin developed inside of tinymist repository, while we were only providing a sample plugin to integrate tinymist into LazyVim. I have a few questions:
- What's the position of the new plugin? Does it only replace the typst-preview.nvim?
- How do the replaced plugin authors think of it? Is it common or supported in (neo)vim community? Will they join or against to build an out-of-box new typst.nvim? We must at least discuss about it with chomosuke.

Reply:
If we could, imo we should push changes to existing plugins to not force people to leave their old settings but still get benefit from our work.

Reply:
Yes I agree that sounds good. I was wondering at the status of all of that, good to know progress is underway. I am not sure on how that integration will work I was assuming you had some deal what that plugin author. To me it makes sense to not have an external preview tool when there is potential for the lsp to have that feature, but I think the whole issue requires discussion.